### DISCUSSION TITLE
Switch state inversion upon start.

### DISCUSSION THREAD

**Discussion by saurav-attri** [arendst/Tasmota#10799] [Q&A] [answered]
Hello!
I am trying to read an external switch's state and broadcast it over MQTT using an ESP01. GPIO2 is in a pull-up configuration with the switch to detect it. 

![Capture](https://user-images.githubusercontent.com/46284122/106580608-1ba90d00-6568-11eb-85a8-bf866c2f27b6.JPG)

I am delaying the onset of the LOW signal at GPIO2 during startup in the case that the external switch is already closed to allow for the ESP01 to boot normally. It works as intended, but the telemetry of the SWITCH's state is inverted if the external switch is already closed upon boot.

Confused about the response, and suspicious of the role of GPIO2 to core functionalities of the ESP module, I tried the same with a nodeMCU and used its GPIO4 as a SWITCH input(without the delay capacitor) and could observe the same inversion.

Is there something I am missing in the Tasmota configuration? Or is it something to be attributed to the ESP module itself?

**Comment by barbudor** [comment, marked as answer]
Which Tasmota version are you using ?
I remember an issue where the value reported in SENSOR was delayed by 1 message and it was fixed in 9.2.0

It could be also that you are using the "wrong" switchmode for your use case. Switchmodes often go by pair with "normal" and an "inverted" one.
Please check  https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

**Reply by saurav-attri** [reply]
Thanks a lot! 
Switchmode was the case, I was using the default '0' toggle mode. Instead, I required Switchmode '1' follow mode to reflect the external switch's state. I am using Tasmota 9.2.0.
